server/src/main/java/io/crate/metadata/doc/DocTableInfo.java

```diff
@@ -683,6 +683,9 @@ public DocTableInfo dropColumns(List<DropColumn> columns) {
         for (var column : columns) {
             ColumnIdent columnIdent = column.ref().column();
             Reference reference = references.get(columnIdent);
+            if (toDrop.contains(reference)) {
```

Comment:
Can you merge this with an `||` with the if below?

Reply:
Thanks for reviewing. I think this cannot be merged since we should not check for `ifExists` and throw `ColumnUnknownException` for references already in `toDrop`(in this case, the child is already added to `toDrop` while dropping the parent).